client/pdf_generator.py

```diff
@@ -0,0 +1,186 @@
+		else:
+			self.logger.info('pdf attachement made linking with uid: ' + target.uid)
+
+	def get_template(self, url):
+		logo = self.config['logo']
+		formatted_time = time.ctime()
+		company = self.application.config['mailer.company_name']
+		sender = self.application.config['mailer.source_email_alias']
+
+		story = []
+		click_me = self.config['link_text']
+		link = '<font color=blue><link href="' + str(url) + '">' + click_me + '</link></font>'
+		if self.config['logo']:
+			img = platypus.Image(logo, 2 * inch, inch)
+			story.append(img)
+
+		style_sheet = styles.getSampleStyleSheet()
+		style_sheet.add(styles.ParagraphStyle(name='Justify', alignment=TA_JUSTIFY))
+		ptext = '<font size=10>' + formatted_time + '</font>'
+		story.append(platypus.Spacer(1, 12))
+		story.append(platypus.Paragraph(ptext, style_sheet['Normal']))
+		story.append(platypus.Spacer(1, 12))
+		with open(self.config['template_file'], 'r') as file_h:
+			for line in file_h:
+				story.append(platypus.Paragraph(line, style_sheet['Normal']))
+		story.append(platypus.Spacer(1, 8))
+		story.append(platypus.Paragraph(link, style_sheet['Justify']))
+		story.append(platypus.Spacer(1, 12))
+		ptext = '<font size=10>Sincerely,</font>'
+		story.append(platypus.Paragraph(ptext, style_sheet['Normal']))
+		story.append(platypus.Spacer(1, 12))
+		ptext = '<font size=10>' + sender + '</font>'
+		story.append(platypus.Paragraph(ptext, style_sheet['Normal']))
+		story.append(platypus.Spacer(1, 12))
+		ptext = '<font size=10>' + company + '</font>'
+		story.append(platypus.Paragraph(ptext, style_sheet['Normal']))
+		return story
+
+	def attach_pdf(self, outfile):
+		self.application.config['mailer.attachment_file'] = outfile
+
+	def signal_send_finished(self, _):
+		if not os.path.isfile(self.config['output_pdf']) and os.access(self.config['output_pdf']. os.W_OK):
+			self.logger.error('no pdf file found at: ' + str(self.config['output_pdf']))
+			return
+		self.logger.info('deleting pdf file: ' + str(self.config['output_pdf']))
+		try:
+			os.remove(self.config['output_pdf'])
+		except Exception as err:
+			self.logger.debug('cannot delete created attachment: ' + repr(err))
+		self.application.config['mailer.attachment_file'] = None
+
+	def expand_path(self, outfile, *args, **kwargs):
+		expanded_path = _expand_path(outfile, *args, **kwargs)
+		try:
+			expanded_path = mailer.render_message_template(expanded_path, self.application.config)
+		except jinja2.exceptions.TemplateSyntaxError as error:
+			self.logger.error("jinja2 syntax error ({0}) in directory: {1}".format(error.message, outfile))
+			self.text_insert("Jinja2 syntax error ({0}) in directory: {1}\n".format(error.message, outfile))
+			return None
+		except ValueError as error:
+			self.logger.error("value error ({0}) in directory: {1}".format(error, outfile))
+			self.text_insert("Value error ({0}) in directory: {1}\n".format(error, outfile))
+			return None
+		return expanded_path
+		
```

Comment:
There's some trailing spaces that were somehow added down here. The file should end with a single blank line with no whitespace on it.